### PR TITLE
fix: expire sandboxes immediately when shutdownTime is reached

### PR DESCRIPTION
**Pull request by rayowang**
fixes #543 

## Summary

This PR fixes the sandbox expiry boundary at the exact shutdown time.

## Root cause

`checkSandboxExpiry` used a strict `After` comparison, so the exact equality case (`now == shutdownTime`) was not considered expired.

## Changes

- treat `shutdownTime <= now` as expired
- add deterministic test coverage for the exact equality boundary
- keep coverage for the minimum requeue behavior in the near-future case

### CODE REVIEW

**Comment by rayowang** [comment]
@soltysh @barney-s Please help review and approve, thanks! cc @janetkuo @justinsb

File: controllers/sandbox_controller.go [hunk 1]
```diff
@@ -811,17 +811,21 @@ func (r *SandboxReconciler) handleSandboxExpiry(ctx context.Context, sandbox *sa
 // returns true if expired, false otherwise
 // if not expired, also returns the duration to requeue after
 func checkSandboxExpiry(sandbox *sandboxv1alpha1.Sandbox) (bool, time.Duration) {
+	return checkSandboxExpiryAt(sandbox, time.Now())
+}
+
+func checkSandboxExpiryAt(sandbox *sandboxv1alpha1.Sandbox, now time.Time) (bool, time.Duration) {
```
Comment (barney-s) [inline controllers/sandbox_controller.go]: is there a reason to add a new method instead of changing the old method ?

Comment (rayowang) [reply controllers/sandbox_controller.go]: @barney-s  I originally added a separate helper only to inject `now` for deterministic tests while keeping the production call site unchanged.

I agree that the extra method wasn’t really necessary here, so I simplified it by changing the existing `checkSandboxExpiry` method to accept `now` directly and updated the caller/tests accordingly.

**Comment by rayowang** [comment]
@barney-s @aditya-shantanu I’ve made the updates. I’ve made the updates. Could you please review it again?

**Comment by barney-s** [comment]
Please rebase the PR there seems to be a conflict @rayowang

**Comment by aditya-shantanu** [comment]
@rayowang PR needs rebase :-(

**Comment by rayowang** [comment]
@aditya-shantanu > @rayowang PR needs rebase :-(

Rebase done. Please take another look, thanks!
